websocket/auth_ws_channel.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023-present Coinbase Global, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import asyncio, base64, hashlib, hmac, json, os, time, websockets
+
+API_KEY = str(os.environ.get('API_KEY'))
+PASSPHRASE = str(os.environ.get('PASSPHRASE'))
+SECRET_KEY = str(os.environ.get('SECRET_KEY'))
+
+URI = 'wss://ws-feed.exchange.coinbase.com'
+SIGNATURE_PATH = '/users/self/verify'
+
+channel = 'level2'
+product_ids = 'ETH-USD'
+
+
+async def generate_signature():
+    timestamp = str(time.time())
+    message = f'{timestamp}GET{SIGNATURE_PATH}'
+    hmac_key = base64.b64decode(SECRET_KEY)
+    signature = hmac.new(
+        hmac_key,
+        message.encode('utf-8'),
+        digestmod=hashlib.sha256).digest()
+    signature_b64 = base64.b64encode(signature).decode().rstrip('\n')
+    return API_KEY, SECRET_KEY, PASSPHRASE, signature_b64, timestamp
```

Comment:
you can drop SECRET_KEY as a return value

Reply:
actually i would simplify this to only signature_b64 and timestamp. api_key and passphrase can be used as the global constants above